File: CS/WinSolution.Module/Classes2.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ConditionalAppearance;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WinSolution.Module {

    [NavigationItem]
    [Appearance("Can Edit Items", TargetItems = "Items", Criteria = "!CanEdit", Enabled = false)]
    public class OwnerClass : BaseObject {
        public OwnerClass(Session s) : base(s) { }
        private bool _CanEdit;
        [ImmediatePostData]
        public bool CanEdit {
            get { return _CanEdit; }
            set { SetPropertyValue<bool>("CanEdit", ref _CanEdit, value); }
        }
        [Aggregated]
        [Association]
        public XPCollection<ItemCLass> Items {
            get { return GetCollection<ItemCLass>("Items"); }
        }
    }

    [DefaultListViewOptions(true, DevExpress.ExpressApp.NewItemRowPosition.None)]
    [Appearance("Can Edit Value", TargetItems = "Value", Criteria = "not (Owner.CanEdit and CanEdit)", Enabled = false)]
    public class ItemCLass : HCategory {
        public ItemCLass(Session s) : base(s) { }
        private OwnerClass _Owner;
        [Association]
        public OwnerClass Owner {
            get { return _Owner; }
            set { SetPropertyValue<OwnerClass>("Owner", ref _Owner, value); }
        }
        private bool _CanEdit;
        [ImmediatePostData]
        public bool CanEdit {
            get { return _CanEdit; }
            set { SetPropertyValue<bool>("CanEdit", ref _CanEdit, value); }
        }
        private int _Value;
        public int Value {
            get { return _Value; }
            set { SetPropertyValue<int>("Value", ref _Value, value); }
        }
    }

}

File: VB/WinSolution.Module/Classes2.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ConditionalAppearance
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace WinSolution.Module

    <NavigationItem, Appearance("Can Edit Items", TargetItems := "Items", Criteria := "!CanEdit", Enabled := False)> _
    Public Class OwnerClass
        Inherits BaseObject

        Public Sub New(ByVal s As Session)
            MyBase.New(s)
        End Sub
        Private _CanEdit As Boolean
        <ImmediatePostData> _
        Public Property CanEdit() As Boolean
            Get
                Return _CanEdit
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("CanEdit", _CanEdit, value)
            End Set
        End Property
        <Aggregated, Association> _
        Public ReadOnly Property Items() As XPCollection(Of ItemCLass)
            Get
                Return GetCollection(Of ItemCLass)("Items")
            End Get
        End Property
    End Class

    <DefaultListViewOptions(True, DevExpress.ExpressApp.NewItemRowPosition.None), Appearance("Can Edit Value", TargetItems := "Value", Criteria := "not (Owner.CanEdit and CanEdit)", Enabled := False)> _
    Public Class ItemCLass
        Inherits HCategory

        Public Sub New(ByVal s As Session)
            MyBase.New(s)
        End Sub
        Private _Owner As OwnerClass
        <Association> _
        Public Property Owner() As OwnerClass
            Get
                Return _Owner
            End Get
            Set(ByVal value As OwnerClass)
                SetPropertyValue(Of OwnerClass)("Owner", _Owner, value)
            End Set
        End Property
        Private _CanEdit As Boolean
        <ImmediatePostData> _
        Public Property CanEdit() As Boolean
            Get
                Return _CanEdit
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("CanEdit", _CanEdit, value)
            End Set
        End Property
        Private _Value As Integer
        Public Property Value() As Integer
            Get
                Return _Value
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("Value", _Value, value)
            End Set
        End Property
    End Class

End Namespace
